frontend/dockerfile/dockerfile2llb/convert_runmount.go

```diff
@@ -113,12 +114,12 @@ func dispatchRunMounts(d *dispatchState, c *instructions.RunCommand, sources []*
 				sharing = llb.CacheMountLocked
 			}
 			if mount.CacheID == "" {
-				mount.CacheID = path.Clean(mount.Target)
+				mount.CacheID = filepath.Clean(mount.Target)
```

Comment:
`filepath` in here doesn't look right as this code can run in a different OS than daemon or the `mount.Target`

Reply:
This was my attempt to have the Windows path resolve to the same default cache ID e.g.
```
C:/temp -> C:\\temp
```
though this quickly falls apart with cases like `/temp -> \\temp`.

Best guarantee is just for the user to set their own IDs if they are using different path schemes. Will add that as part of the documentation note.

Let me revert back to `path.Clean`.